news-frontend/src/utils/MainApi.js

```diff
@@ -0,0 +1,80 @@
+class MainApi {
+  constructor(baseUrl, headers) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error ${res.status}`);
+  }
+
+  saveArticle (article, token) {
+    return fetch(`${this._baseUrl}/articles`, {
+      method: "POST",
+      headers: {
+        ...this._headers,
+        Authorization: `Bearer ${token}`,
+      },
+      body: JSON.stringify(article),
+    }).then(this._checkResponse);
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.